src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceProviderContainerTests.cs

```diff
@@ -494,6 +577,7 @@ public async Task GetRequiredService_UsesSingletonAndLazyLocks_NoDeadlock()
             Assert.Equal("1234", sb.ToString()); // Expected order of execution
         }
 
+        private ManualResetEvent _mreForThread3 = new ManualResetEvent(false);
```

Comment:
I think it would be better to declare these `ManualResetEvent` fields as local variables in the test method themselves. And just pass them through the lambda methods as closures. It is a bit confusing to have fields on the test class that some time get used in some test methods.